lib/player.js

```diff
@@ -1694,17 +1694,24 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     // initialize it now even through it appears a little out-of-place.
     this.regionTimeline_ =
         new shaka.media.RegionTimeline(() => this.seekRange());
-    this.regionTimeline_.setListeners(/* onRegionAdded= */ (region) => {
+    this.regionTimeline_.addEventListener('regionadd', (event) => {
+      /** @type {shaka.extern.TimelineRegionInfo} */
+      const region = event['region'];
       this.onRegionEvent_(shaka.Player.EventName.TimelineRegionAdded, region);
+
       if (this.adManager_) {
         this.adManager_.onDashTimedMetadata(region);
       }
     });
+
     this.qualityObserver_ = null;
     if (this.config_.streaming.observeQualityChanges) {
       this.qualityObserver_ = new shaka.media.QualityObserver(
           () => this.getBufferedInfo());
-      this.qualityObserver_.setListeners((mediaQualityInfo, position) => {
+
```

Comment:
nit; extra \n.

Reply:
That's intentional.  I found it easier to read with a little extra whitespace.  I think of it as two "paragraphs" of code.